### ISSUE TITLE
Blocked aria-hidden on an element because its descendant retained focus.

### ISSUE THREAD

**Issue by JohnCido** [kobaltedev/kobalte#541]
**Describe the bug**
DropdownMenu print this error when opened first time on a page:

```log
Blocked aria-hidden on an element because its descendant retained focus. The focus must not be hidden from assistive technology users. Avoid using aria-hidden on a focused element or its ancestor. Consider using the inert attribute instead, which will also prevent focus. For more details, see the aria-hidden section of the WAI-ARIA specification at https://w3c.github.io/aria/#aria-hidden.
Element with focus: button
Ancestor with aria-hidden:  <div id="app" aria-hidden="true">...
```

**To Reproduce**

1. Go to [official document page](https://kobalte.dev/docs/core/components/dropdown-menu/)
2. Open DevTool
3. Click on first example button
4. Check error in the console

**Expected behavior**
No error should be printed.

**Screenshots**
<img width="1401" alt="截屏2024-12-25 15 17 54" src="https://github.com/user-attachments/assets/77bba4de-5aa2-4c6f-b6a3-de1f03612af5" />

**Desktop (please complete the following information):**

- OS: macOS
- Browser Chrome
- Version 131.0.6778.205

**Comment by Kang-Jh** [comment]
I saw the same error console too when I'm using `<Dialog>`. But I cannot reproduce it. I guess there's some common behavior between dialog and drop down 🤔

**Comment by wontheone1** [comment]
Yes, I am experiencing the same problem so I can +1 on this issue!

**Comment by Azq2** [comment]
some shit with dialog

**Comment by axelra82** [comment]
+1

**Comment by Bravilogy** [comment]
+1 with `<Select>`.

**Comment by daveycodez** [comment]
Still reproduces on `@kobalte/core@0.13.12`. Sharing a root-cause analysis since the thread is otherwise +1s — this affects every modal overlay (`DropdownMenu`, `Dialog`, `Select`, `Popover`…) because they all share the same `createHideOutside` path.

### Root cause: aria-hide runs before focus lands inside the overlay

On open, two things happen in the same reactive flush:

1. **`createHideOutside`** ([`create-hide-outside.ts`](https://github.com/kobaltedev/kobalte/blob/main/packages/core/src/primitives/create-hide-outside/create-hide-outside.ts)) runs inside a plain `createEffect`, so during the **effect phase** it *synchronously* walks from `<body>` and `setAttribute("aria-hidden", "true")` on everything outside the content — `#app` (which contains the trigger) included.
2. The overlay's focus move is **deferred** — `createSelectableList({ autoFocus, deferAutoFocus: true })` in the menu/listbox content, plus `createFocusScope`'s mount autofocus. So focus is still on the trigger when step 1 fires.

Result: for one tick the still-focused trigger sits under an `aria-hidden` ancestor, and Chromium logs *"Blocked aria-hidden … descendant retained focus."* Chrome actually **blocks** the attribute while focus is inside, so AT isn't harmed, and a tick later focus lands in the overlay and the hide applies — but the warning is emitted. It shows **once per element per session** because Chrome dedupes it against the (unchanged) `#app` node, which is why it "only happens the first time".

### Why it's ordering, and how React Aria avoids it

React Aria runs the equivalent focus-containment in a **layout effect** (`FocusScope`) that moves focus into the overlay *before* `ariaHideOutside` (a normal effect) runs — layout effects fire first, so focus is already inside the target when the rest of the page is hidden. Kobalte's `deferAutoFocus` inverts that order.

### Possible fixes

- Move focus into the content **before** (or synchronously with) `ariaHideOutside`, rather than deferring it — e.g. run the hide only after the focus scope has taken focus.
- Or hide outside content with the **`inert` attribute** instead of `aria-hidden` (as the warning itself suggests) — `inert` both hides from AT and removes focusability, and doesn't trip this heuristic.
- Or exclude the currently-focused trigger from the hidden set until focus has moved.

Happy to help test a fix. Minimal repro is the [official DropdownMenu docs](https://kobalte.dev/docs/core/components/dropdown-menu/) — open DevTools, click the first trigger.

**Comment by Azq2** [comment]
@jer3m01 great thanks
